source/common/router/router.cc

```diff
@@ -1571,6 +1634,11 @@ void Filter::onUpstreamHeaders(uint64_t response_code, Http::ResponseHeaderMapPt
                                UpstreamRequest& upstream_request, bool end_stream) {
   ENVOY_STREAM_LOG(debug, "upstream headers complete: end_stream={}", *callbacks_, end_stream);
 
+  if (host_selection_cancelable_) {
```

Comment:
OOC: How would we get headers if we're still waiting to choose a host?

Reply:
outdated - local replies are handled by onLocalReply